controller/util/cf_v2_client.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	model "github.com/kubernetes-incubator/service-catalog/model/service_broker"
+	scmodel "github.com/kubernetes-incubator/service-catalog/model/service_controller"
+)
+
+const (
+	catalogFormatString         = "%s/v2/catalog"
+	serviceInstanceFormatString = "%s/v2/service_instances/%s"
+	bindingFormatString         = "%s/v2/service_instances/%s/service_bindings/%s"
+)
+
+type cfV2BrokerClient struct {
+	broker *scmodel.ServiceBroker
+}
+
+// Verify that client implements the BrokerClient interface.
+var _ BrokerClient = (*cfV2BrokerClient)(nil)
+
+// CreateCFV2BrokerClient creates an instance of BrokerClient for communicating
+// with brokers which implement the Cloud Foundry Service Broker V2 API.
+func CreateCFV2BrokerClient(b *scmodel.ServiceBroker) BrokerClient {
+	return &cfV2BrokerClient{
+		broker: b,
+	}
+}
+
+func (c *cfV2BrokerClient) GetCatalog() (*model.Catalog, error) {
+	url := fmt.Sprintf(catalogFormatString, c.broker.BrokerURL)
+
+	req, err := http.NewRequest("GET", url, nil)
+	req.SetBasicAuth(c.broker.AuthUsername, c.broker.AuthPassword)
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
Was staying away from actually modifying the code, but this makes sense. I've opted to just create an HTTP client in the Create() method to keep it abstracted rather than require dependency injection. Let me know if you prefer the other.